crates/printer/src/standard.rs

```diff
@@ -1360,6 +1375,13 @@ impl<'a, M: Matcher, W: WriteColor> StandardImpl<'a, M, W> {
     ) -> io::Result<()> {
         if self.config().max_columns_preview {
             let original = line;
+            if let Some(p) = self.config().max_columns_preview_before {
+                let start = matches[0].start().saturating_sub(p as usize);
+                if start != 0 {
+                    self.write(b"[... omitted start of long line] ")?;
+                    line = line.with_start(start);
+                }
+            }
```

Comment:
Compare to the full `--column-context-before` + `--column-context-after` support that needs to modify everywhere, the implementation code here is quiet simple.